File: WebShopKBS/WebShopKBS/Controllers/EmployeeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using WebShopKBS.Models;
using WebShopKBS.Services;
using WebShopKBS.UoW;

namespace WebShopKBS.Controllers
{
	[RoutePrefix("employee")]
    public class EmployeeController : ApiController
    {
	    private readonly EmployeeService service;

	    public EmployeeController()
	    {
		    service = new EmployeeService(new UnitOfWork());
	    }

	    [Route("updateOrder")]
	    [HttpPost]
	    public IHttpActionResult UpdateOrder([FromBody] Order order)
	    {
		    var returnOrder = service.UpdateOrder(order);
		    if (returnOrder != null)
			    return Ok();
		    return BadRequest("Order not Found");
	    }

	    [Route("restock")]
	    [HttpPost]
	    public IHttpActionResult Restock([FromBody] List<Item> restockList)
	    {
		    try
		    {
			    service.RequestRestock(restockList);
			    return Ok();
		    }
		    catch (Exception)
		    {
			    return BadRequest();
		    }
	    }

	    [Route("orders")]
	    [HttpGet]
	    public IEnumerable<Order> GetOrders()
	    {
		    return service.GetOrders();
	    }

	    [Route("items")]
	    [HttpGet]
	    public IEnumerable<Item> GetItems()
	    {
		    return service.GetItems();
	    }
	}
}

File: WebShopKBS/WebShopKBS/App/Controllers/EmployeeController.js
﻿angular.module('employee.controllers', [])
.controller('employeeController',
['$scope', '$window', '$state', 'employeeService',
	function ($scope, $window, $state, employeeService) {
		$scope.addToList = function(item) {
			$scope.restockItems.push(item);
		}

		$scope.removeFromList = function(item) {
			var index = $scope.restockItems.indexOf(item);
			$scope.restockItems[index].splice(index, 1);
		}

		$scope.getOrders = function () {
			employeeService.getOrders().then(function(response) {
				$scope.orders = response.data;
			});
		}

		$scope.getItems = function () {
			employeeService.getItems().then(function (response) {
				$scope.items = response.data;
			});
		}

		$scope.updateOrder = function (order) {
			employeeService.updateOrder(order).then(function (response) {
				alert('Order is updated');
			});
		}

		$scope.restock = function () {
			employeeService.restock($scope.restockItems).then(function (response) {
				$window.location.reload();
			});
		}

		var getContent = function() {
			$scope.getOrders();
			$scope.getItems();
		}

		getContent();
	}])